src/capabilites/index.ts

```diff
@@ -1,16 +1,52 @@
-import { UiAppCapabilityType, UiAppEventType } from '../constants';
+import {
+    UiAppCapabilityType,
+    UiAppEventToSubscribeType,
+    UiAppEventToTriggerType
+} from '../constants';
 
 import { CapabilityManager } from './capabilityManager';
 
 export type { CapabilityManager } from './capabilityManager';
 
 class DashboardCogMenuManager extends CapabilityManager {
     type = UiAppCapabilityType.DASHBOARD_COG_MENU;
-    events = [UiAppEventType.DASHBOARD_COG_MENU_CONTEXT];
+    eventsToSubscribe = [UiAppEventToSubscribeType.DASHBOARD_COG_MENU_CONTEXT];
+    eventsToTrigger = [];
 
     getAdditionalClientMethods() {
         return {};
     }
 }
 
-export const capabilityManagers = [DashboardCogMenuManager];
+class AppRoutingManager extends CapabilityManager {
+    type = UiAppCapabilityType.APP_ROUTING;
+    eventsToSubscribe = [];
+    eventsToTrigger = [
+        UiAppEventToTriggerType.RELOAD_FRAME,
+        UiAppEventToTriggerType.OPEN_URL
```

Comment:
I'm thinking maybe we just add an `allow-popups` sandbox attribute to the iframes so that they can contain `target="_blank"` links? That seems simpler than requiring that they use an event. It could potentially mean some apps could get annoying with many popups, but the user can always uninstall. I don't think there's a security risk. I definitely see a need for a different `navigate` event, in which the iframe requests that the entire app navigate to a new page (not in a new tab). The usecase would be for programatically navigating somewhere else within the datadog app, like to a certain infrastructure view or to a custom page when we add that capability

Reply:
that was my first approach actually, but it turns out even with `allow-popups`, the new page will have limitation, for example you can't submit forms. This approach brings the control back to datadog which allows us to expand on it in the future to allow white/black listing of urls and can evolve into navigating routes within the same app or different apps. This made more sense to me.

Reply:
We should definitely enable `allow-forms`, that should handle the case you're mentioning. Anyways, thanks for explaining. We'll see how it goes,, if it turns out to be a huge hassle for developers we can turn it on later